sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2491,6 +2518,14 @@ object GpuOverrides {
         override val childExprs: Seq[BaseExprMeta[_]] =
           rp.ordering.map(GpuOverrides.wrapExpr(_, conf, Some(this)))
 
+        override def tagPartForGpu() {
+          val numPartitions = rp.numPartitions
+          if (numPartitions > 1) {
```

Comment:
This needs to be combined with a check for struct/nested type in the `rp.ordering`

```
if (numPartitions > 1 && re.ordering.exists(so => isNestedType(so.dataType))) {
...
```

Reply:
you are right